packages/components/pill/src/Pill.tsx

```diff
@@ -50,6 +52,18 @@ export const Pill = React.forwardRef<HTMLDivElement, ExpandProps<PillProps>>(
     } = props;
 
     const styles = getPillStyles(variant);
+    const [textIsEllipsis, setTextIsEllipsis] = React.useState(false);
```

Comment:
I'd prefer a clearer name here so that someone who's new to the project could understand what it means. Something like `textIsTruncated` maybe, or `textOverflowsContainer`.